src/Interpreters/TreeRewriter.cpp

```diff
@@ -1363,7 +1363,10 @@ void TreeRewriter::normalize(
     MarkTableIdentifiersVisitor(identifiers_data).visit(query);
 
     /// Rewrite function names to their canonical ones.
-    if (settings.normalize_function_names)
+    /// Notice: function name normalization is disabled when it's a secondary query, because queries are either
+    /// already normalized on initiator node, or not normalized and should remain unnormalized for
+    /// compatibility.
+    if (context->getClientInfo().query_kind != ClientInfo::QueryKind::SECONDARY_QUERY && settings.normalize_function_names)
```

Comment:
It's not the only place where this setting is used

Reply:
I'm not sure if we should skip the whole `TreeRewriter::normalize` if query is secondary. @tavplubix  Do you think it is ok ?

Reply:
Why do we need to skip the whole `TreeRewriter::normalize`?

Reply:
> It's not the only place where this setting is used

Sorry, I misunderstood. Get it now.

Reply:
> It's not the only place where this setting is used

@tavplubix  done.